Memory leak?

We have found this in one of our components which uses latest pushy:
```
LEAK: ReferenceCountedOpenSslContext.release() was not called before it's garbage-collected. Enable advanced leak reporting to find out where the leak occurred. To enable advanced leak reporting, specify the JVM option '-Dio.netty.leakDetection.level=advanced' or call ResourceLeakDetector.setLevel() See http://netty.io/wiki/reference-counted-objects.html for more information.

Stacktrace (most recent call first):

    at Slf4JLogger.java.error(Slf4JLogger.java:176)
    at ResourceLeakDetector.java.reportUntracedLeak(ResourceLeakDetector.java:313)
    at ResourceLeakDetector.java.reportLeak(ResourceLeakDetector.java:289)
    at ResourceLeakDetector.java.track0(ResourceLeakDetector.java:248)
    at ResourceLeakDetector.java.track(ResourceLeakDetector.java:237)
    at ReferenceCountedOpenSslContext.java.<init>(ReferenceCountedOpenSslContext.java:239)
    at ReferenceCountedOpenSslContext.java.<init>(ReferenceCountedOpenSslContext.java:219)
    at ReferenceCountedOpenSslClientContext.java.<init>(ReferenceCountedOpenSslClientContext.java:59)
```

We'll try to enable the advanced leak detection, but anyhow... anyone else has seen this in latest pushy?

In the same thread and in almost same time, we shutdown the 'old' client and create a new one. This is an edited 'log' from the component that shows the leak alert:
```
[2017-06-12T06:36:06.429Z] [INFO] [EventProcessing-8-push] com.mycompany.MyAPNSClientCode:? Sending push message to APNS server
[2017-06-12T06:36:06.430Z] [INFO] [EventProcessing-8-push] com.mycompany.MyAPNSClientCode:? Certificate changed (development=true, oldVersion=26176, newVersion=26186)
[2017-06-12T06:36:06.430Z] [INFO] [EventProcessing-8-push] com.mycompany.MyAPNSClientCode:? Releasing APNS client (development=true, version=26176, hadProblems=false)
[2017-06-12T06:36:06.430Z] [INFO] [EventProcessing-8-push] com.mycompany.MyAPNSClientCode:? Shutting down APNS client (development=true)
[2017-06-12T06:36:06.430Z] [INFO] [EventProcessing-8-push] com.turo.pushy.apns.ApnsClient:? Disconnecting.
[2017-06-12T06:36:06.430Z] [INFO] [EventProcessing-8-push] com.mycompany.MyAPNSClientCode:? Initializing APNS client (development=true)
[2017-06-12T06:36:06.432Z] [INFO] [nioEventLoopGroup-2-4] com.mycompany.MyAPNSClientCode:? ApnsClient shutdown properly (development=true)
[2017-06-12T06:36:06.437Z] [INFO] [EventProcessing-8-push] com.turo.pushy.apns.SslUtil:? Native SSL provider is available and supports ALPN; will use native provider.
[2017-06-12T06:36:06.437Z] [ERROR] [EventProcessing-8-push] io.netty.util.ResourceLeakDetector:? LEAK: ReferenceCountedOpenSslContext.release() was not called before it's garbage-collected. Enable advanced leak reporting to find out where the leak occurred. To enable advanced leak reporting, specify the JVM option '-Dio.netty.leakDetection.level=advanced' or call ResourceLeakDetector.setLevel() See http://netty.io/wiki/reference-counted-objects.html for more information.
```

--- a/pushy/src/main/java/com/turo/pushy/apns/SslUtil.java
+++ b/pushy/src/main/java/com/turo/pushy/apns/SslUtil.java
@@ -49,7 +49,7 @@ class SslUtil {
         if (OpenSsl.isAvailable()) {
             if (OpenSsl.isAlpnSupported()) {
                 log.info("Native SSL provider is available and supports ALPN; will use native provider.");
-                sslProvider = SslProvider.OPENSSL_REFCNT;
+                sslProvider = SslProvider.OPENSSL;
             } else {
                 log.info("Native SSL provider is available, but does not support ALPN; will use JDK SSL provider.");
                 sslProvider = SslProvider.JDK;

